src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -131,13 +140,15 @@ bool HasMoreToSync()
                     _logger.Trace(
                         $"Downloading blocks from peer. CurrentNumber: {currentNumber}, BeaconPivot: {_beaconPivot.PivotNumber}, BestPeer: {bestPeer}, HeaderToRequest: {headersToRequest}");
 
-                BlockHeader[]? headers = _chainLevelHelper.GetNextHeaders(headersToRequest);
-                if (headers == null || headers.Length == 0)
-                    break;
+                // Kind hacky way to forward process blocks that was sent via NewPayload but not available via sync.
+                // Needed by at least 10 "Invalid Ancestor Chain Re-Org.*sync" hive test.
+                // TODO: Probably should not do forward sync/processing in here in addition to having parallel download capability
+                Dictionary<Keccak, Block> knownBlocks = GetExistingBlocks(downloadReceipts, headers);
```

Comment:
I'm wondering if we still need this. I thought new hive tests are working by P2P, but I might be wrong ofc

Reply:
The peer is missing the last block.

Reply:
The peer is missing the last block.